test/parachain/tests/integration_test.rs

```diff
@@ -0,0 +1,404 @@
+
+	State {
+		#[rpc(method = "state_getRuntimeVersion")]
+		fn runtime_version() -> RuntimeVersion;
+	}
+
+	System {
+		#[rpc(method = "system_networkState")]
+		fn network_state() -> Value;
+	}
+}
+
+// Adapted from
+// https://github.com/rust-lang/cargo/blob/485670b3983b52289a2f353d589c57fae2f60f82/tests/testsuite/support/mod.rs#L507
+fn target_dir() -> PathBuf {
+	env::current_exe()
+		.ok()
+		.map(|mut path| {
+			path.pop();
+			if path.ends_with("deps") {
+				path.pop();
+			}
+			path
+		})
+		.unwrap()
+}
+
+struct ChildHelper<'a> {
+	name: String,
+	child: &'a mut Child,
+	stdout: String,
+	stderr: String,
+}
+
+impl<'a> Drop for ChildHelper<'a> {
+	fn drop(&mut self) {
+		let name = self.name.clone();
+
+		self.terminate();
+		eprintln!(
+			"process '{}' stdout:\n{}\n",
+			name,
+			self.read_stdout_to_end().unwrap_or_default()
+		);
+		eprintln!(
+			"process '{}' stderr:\n{}\n",
+			name,
+			self.read_stderr_to_end().unwrap_or_default()
+		);
+	}
+}
+
+impl<'a> ChildHelper<'a> {
+	fn new(name: &str, child: &'a mut Child) -> ChildHelper<'a> {
+		ChildHelper {
+			name: name.to_string(),
+			child,
+			stdout: Default::default(),
+			stderr: Default::default(),
+		}
+	}
+
+	fn read_stdout_to_end(&mut self) -> io::Result<&str> {
+		let mut output = String::new();
+
+		self.child
+			.stdout
+			.as_mut()
+			.ok_or_else(|| io::Error::new(io::ErrorKind::Other, "stdout not captured"))?
+			.read_to_string(&mut output)?;
+		self.stdout.push_str(output.as_str());
+
+		Ok(&self.stdout)
+	}
+
+	fn read_stderr_to_end(&mut self) -> io::Result<&str> {
+		let mut output = String::new();
+
+		self.child
+			.stderr
+			.as_mut()
+			.ok_or_else(|| io::Error::new(io::ErrorKind::Other, "stderr not captured"))?
+			.read_to_string(&mut output)?;
+		self.stderr.push_str(output.as_str());
+
+		Ok(&self.stderr)
+	}
+
+	fn terminate(&mut self) {
+		match self.child.try_wait() {
+			Ok(Some(_)) => return,
+			Ok(None) => {}
+			Err(err) => {
+				eprintln!("could not wait for child process to finish: {}", err);
+				let _ = self.child.kill();
+				let _ = self.child.wait();
+				return;
+			}
+		}
+
+		#[cfg(unix)]
+		{
+			use nix::sys::signal::{kill, Signal::SIGTERM};
+			use nix::unistd::Pid;
+
+			kill(Pid::from_raw(self.child.id().try_into().unwrap()), SIGTERM).unwrap();
+
+			let mut tries = 30;
+
+			let success = loop {
+				tries -= 1;
+
+				match self.child.try_wait() {
+					Ok(Some(_)) => break true,
+					Ok(None) if tries == 0 => break false,
+					Ok(None) => thread::sleep(Duration::from_secs(1)),
+					Err(err) => {
+						eprintln!("could not wait for child process to finish: {}", err);
+						break false;
+					}
+				}
+			};
+
+			if !success {
+				let _ = self.child.kill();
+			}
+		}
+
+		#[cfg(not(unix))]
+		let _ = self.child.kill();
+
+		let _ = self.child.wait();
+	}
+}
+
+fn tcp_port_is_open<A: net::ToSocketAddrs>(address: A) -> bool {
+	match net::TcpStream::connect(&address) {
+		Ok(_) => true,
+		Err(_) => false,
+	}
```

Comment:
```suggestion
	net::TcpStream::connect(&address).is_ok()
```

Rust docs is your friend.